### PR TITLE
Submit application fix

### PR DESCRIPTION
**Pull request by Adam-W1**
https://digital.dclg.gov.uk/jira/browse/FS-2032

- applications key no longer received from account store
- removed silent error and replaced with a throw 500 error on submit keyErrors
- added a test to check submit application returns expected body
- refactored/tidied tests

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -2,5 +2,5 @@
 env =
     FLASK_ENV=unit_test
     FLASK_DEBUG=1
-    D:DATABASE_URL=postgresql://postgres:postgres@127.0.0.1:5432/postgres
+    D:DATABASE_URL=postgresql://postgres:postgres@127.0.0.1:5432/application_store
```
Comment (gidsg) [inline pytest.ini:5]: please revert this

**Review by gidsg — APPROVED** [review]
LGTM 👍 are we sure we don't have other usages which will be broken by this API changing?

File: pytest.ini (hunk 1 above)
Comment (gidsg) [reply pytest.ini:5]: ...this change is causing tests to fail.

Comment (Adam-W1) [reply pytest.ini]: my bad, removed

**Pull request merged**